numpy/core/records.py

```diff
@@ -767,6 +767,9 @@ def fromfile(fd, dtype=None, shape=None, offset=0, formats=None,
     >>> r.shape
     (10,)
     """
+    
+    if dtype is None and formats is None:
+        raise ValueError("Must have dtype= or formats=")
```

Comment:
Missing arguments are normally treated as a `TypeError`. Perhaps `raise TypeError('fromfile() needs a 'dtype' or 'formats' argument`, which is sort of similar to the builtin `TypeError: foo() missing 1 required keyword-only argument: 'x'` message forgetting a function argument gives

Reply:
I've just copied the check from the _fromstring_ function (https://github.com/numpy/numpy/blob/master/numpy/core/records.py#L722). Should I apply your change in both functions?